mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -491,6 +492,10 @@ pub async fn lp_init_continue(ctx: MmArc) -> MmInitResult<()> {
     #[cfg(target_arch = "wasm32")]
     init_wasm_event_streaming(&ctx);
 
+    // This function spwans related WalletConnect related tasks and needed initialization before
+    // WalletConnect can be usable in KDF.
+    WalletConnectCtx::from_ctx(&ctx).mm_err(|err| MmInitError::WalletInitError(err.to_string()))?;
```

Comment:
Does it make sense to initialize wallet connect ctx here? It should be with platform coin activation. On the other hand, we are looking to add initializing mm2 with wallet connect in the future instead of crypto ctx. What do you think?

Reply:
Does this imply that each platform coin would have its own WC instance? That doesn’t seem like the intended approach, since I believe WC should be initialized a single time and within `lp_init_continue` or else session creations will only  be valid/useable within a `coin type`, multiple connections to wc relay(expensive) and some other cons

> we are looking to add initializing mm2 with wallet connect in the future instead of crypto ctx.

Yeah, I'm aware. Initializing once in `lp_init_continue` looks like the  approach we'll consider when we're ready for this discussion.

Reply:
> Does this imply that each platform coin would have its own WC instance?

No, I meant to initialize it once from global ctx but when a coin that needs it is activated. So a user that doesn't use ETH or Cosmos doesn't initialize it. But I am fine with keeping it as implemented as it will be revisited later when we want to initialize mm2 with wallet connect.

Reply:
ah okay. it's doable